bgpd/bgp_zebra.c

```diff
@@ -2114,6 +2114,25 @@ int bgp_redistribute_unset(struct bgp *bgp, afi_t afi, int type,
 	return CMD_SUCCESS;
 }
 
+int bgp_redistribute_unset(struct bgp *bgp, afi_t afi, int type,
+			   unsigned short instance)
+{
+	int ret = CMD_SUCCESS;
+	struct listnode *node, *nnode;
+	struct bgp_redist *red;
+
+	if (type != ZEBRA_ROUTE_TABLE || instance != 0)
+		return _bgp_redistribute_unset(bgp, afi, type, instance);
+
+	/* walk over instance */
+	if (!bgp->redist[afi][type])
+		return CMD_SUCCESS;
+	for (ALL_LIST_ELEMENTS(bgp->redist[afi][type], node, nnode, red))
+		ret = _bgp_redistribute_unset(bgp, afi, type, red->instance);
```

Comment:
Do we return only the last call? Shouldn't we check what we get from `_bgp_redistribute_unset()` and return earlier if an error?

Reply:
> 

This call with the instance parameter set to 0 means bgp_delete() is called.
Eventually, the return code can be ignored, and a 0 return is sufficient.